### ISSUE TITLE
Add Worktree support for creating a new branch

### ISSUE THREAD

**Issue by jkyoutsey** [gitkraken/vscode-gitlens#1977]
I would like to be able to create a new branch as part of creating a worktree, the equivalent of `git worktree add <path>` for a branch that does not exist.

It would be nice if we could add some configuration around it that would help make this cleaner for mono-repos:

1. It would be nice to be able to specify a default path prefix for worktrees. I want to add my worktrees to `../../worktrees/<new-branch-name>` so that the new tree is outside of my main repo folder. This is just a nice-to-have. As long as I can do something like `>GitLens+: Add new worktree`, `create`, `../../worktrees/path` and it creates the folder and worktree, I'd be thrilled!
2. It would be nice to be able to open the repo in VSCode at a specific folder. I'd like to be able to open the `./ng` folder under the root of my new worktree in VSCode after creating it so I am where I need to be to run all of my CLI commands and don't see parts of the mono-repo that are not relevant to what I'm working on. Not sure how specifying that in the Add Worktree command would work, but it would save having to drop out to Finder after each command to go find the folder and open it in VSCode.

**Comment by eamodio** [comment]
For your first request you should be able to use the following choices from the worktree create menu:

![image](https://user-images.githubusercontent.com/641685/165363168-72520277-281d-4580-a1f1-f603fd46e135.png)

Also using the _GitLens: Open Settings_ command and jumping down to the Worktrees view, you can tweak some of the defaults:

![image](https://user-images.githubusercontent.com/641685/165363604-09f04389-56ff-44cf-a615-c8b982d90062.png)

Even though it says to provide an absolute path, a relative path should work.

As for the second request, I'm looking into a way to have/create a saved workspace file associated with each worktree, which would allow you to customize it (e.g. open to a sub-folder).

**Comment by jkyoutsey** [comment]
@eamodio When I try to create a worktree from this menu, the + button in the WORKTREES view:
![image](https://user-images.githubusercontent.com/3846466/165395223-f1c53e40-2339-4e0a-bd63-ebbecf8399e2.png)

I get this menu that only shows branches and tags:
![image](https://user-images.githubusercontent.com/3846466/165394856-8bce1959-0e45-471e-bdbb-e0f8854f82ab.png)

So, I don't see any options for Create New Branch and Worktree as your screenshot shows

**Comment by eamodio** [comment]
Ah, you need to choose a branch or tag to create the new branch from. So once you pick a branch or tag in your screenshot you'll get the quickpick menu I showed.

**Comment by jkyoutsey** [comment]
Yeah, that's what I'm trying to request. I don't want to pick an existing branch. I want to DEFINE a new branch when creating the work tree. The git worktree command supports this. I'd like GitLens to give me an option for doing it. 

In its simplest form, git worktree add <path> automatically creates a new branch whose name is the final component of <path>, which is convenient if you plan to work on a new topic.

**Comment by eamodio** [comment]
Yes, but that branch must be created FROM some commit (e.g. a branch, tag, etc). The git command by default (if unspecified) creates a new branch from the current HEAD -- and GitLens will do the same if you just pick the current branch from the quickpick (it will be the first one with the ✔️ ). Then you pick that you want to create a new branch, and then it will ask you for a new branch name.

**Comment by eamodio** [comment]
I should probably add a `HEAD` choice in that list to make sure it works even if you are in a detached HEAD state.

**Comment by jkyoutsey** [comment]
Thank you. That's what wasn't clear. That should work.

**Comment by eamodio** [comment]
Sorry for the confusion. I really wish VS Code allowed for dialog/form based UI, that would make it soo much easier to see all the options without needing to reveal each step one at a time.

Will have to spend some more time thinking through the UX to make this clearer.
